src/main.rs

```diff
@@ -55,53 +56,58 @@ use llvm_interpreter::LLVMInterpreter;
 use error::*;
 
 // FIXME: How do you represent the usage style in POSIX notation?
-fn print_usage() {
-    if cfg!(feature = "llvm-backend") {
-        println!(
-            "usage: balloon [--repl-llvm | [MODE] FILE]
 
---repl-llvm     launches the experimental REPL"
-        );
-    } else {
-        println!("usage: balloon [[MODE] FILE]");
-    }
-    println!(
-        "
-where MODE is one of:
---run           (default) runs the file [FILE]
---check         type check the file [FILE]
---parse         only parse the file [FILE], don't run it
-
-Not passing any arguments to balloon will start the REPL."
-    );
-}
+use clap::{Arg, App};
 
 fn main() {
-    let args: Vec<String> = env::args().collect();
-
-    match args.len() {
-        1 => repl::run_repl(AstWalkInterpreter::new()),
-        2 => {
-            match args[1].as_str() {
-                #[cfg(feature = "llvm-backend")]
-                "--repl-llvm" => repl::run_repl(LLVMInterpreter::new()),
-                filepath => run_file(filepath, AstWalkInterpreter::new()),
-            }
-        }
-        3 => {
-            match args[1].as_str() {
-                "--run" => run_file(&args[2], AstWalkInterpreter::new()),
-                "--check" => typecheck_file(&args[2]),
-                "--parse" => {
-                    if let Some(ast) = parse_file(&args[2]) {
-                        println!("{:#?}", ast);
-                    }
-                }
-                _ => print_usage(),
-            };
+    let mut app = App::new("balloon")
+        .version("0.1.0")
+        .arg(Arg::with_name("file")
+            .help("run the file")
```

Comment:
I put `file` to deal with `balloon program.bl` and `run` for `balloon --run program.bl`
They're the same because `run` is the default mode.
If we use something like `--mode`, then I can put `run` as the default value if the user doesn't supply one, but I can't think of another alternative for the current scenario.
What do you suggest?